promotion.md

```diff
@@ -259,6 +224,76 @@ or `const` item and refer to that.
 *Dynamic check.* The Miri engine could dynamically check this by ensuring that
 the result of computing a promoted is a value that does not need dropping.
 
+### Access to a `const` or `static`
+
+When accessing a `const` in a promotable context, its value anyway gets computed
+at compile-time, so we do not have to check the initializer.  However, the
+restrictions described above still apply for the *result* of the promoted
+computation: in particular, it must be a valid `const` (i.e., it cannot
+introduce interior mutability) and it must not require dropping.
```

Comment:
It's worth noting that these restrictions apply *only* in the context lifetime extension. We can initialize an array, which also uses the implicit rules, with a constant containing interior mutability (i.e. `[CELL; 4]`).

Reply:
But `[&CELL; 4]` gets rejected, I hope?

Reply:
That compiles on stable because `&T` is `Copy` for all `T`. Any use of it will cause a borrow check error since the access to `CELL` has a temporary lifetime.

Reply:
> Promotion fails for &CELL even if it's in an array initializer if that's what you're asking.

Good.

But then behavior seems to be the same for arrays and lifetime extension? It's just that lifetime extension *always* involves a borrow.